### PR TITLE
Fix check

### PR DESCRIPTION
**Pull request by D4nte**
`yarn check` != `yarn run check`
Also, we were not doing a check in release process.
Finally, best practice to not check-in generated files.

PS: What is `postintall` https://docs.npmjs.com/misc/scripts

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: yarn install, build, and test
         run: |
           yarn
-          yarn check
+          yarn run check
```
Comment (luckysori) [inline .github/workflows/nodejs.yml]: :mega: I believe you need to run `yarn postinstall` beforehand for this to pass. Without the files inside `gen/` it won't compile.

File: package.json [hunk 2]
```diff
@@ -7,12 +7,10 @@
   "repository": "git@github.com:comit-network/comit-js-sdk.git",
   "author": "CoBloX developers <team@coblox.tech>",
   "scripts": {
-    "check": "tsc --noEmit",
+    "check": "tsc --noEmit && tslint -p . && prettier --check '**/*.{ts,js,json,yml}'",
     "rebuild": "rimraf dist/ && tsc",
-    "gen": "mkdir -p ./gen && json2ts -i ./siren.schema.json -o ./gen/siren.d.ts",
+    "postinstall": "mkdir -p ./gen && json2ts -i ./siren.schema.json -o ./gen/siren.d.ts && prettier --write ./gen/siren.d.ts",
```
Comment (luckysori) [inline package.json]: :question: Why did you call it `postinstall`?

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (D4nte) [reply .github/workflows/nodejs.yml]: `yarn` does `yarn install`.
The `postinstall` script is always ran automatically after `install`.

File: package.json (hunk 2 above)
Comment (D4nte) [reply package.json]: so that it's run automatically after `install`.

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (luckysori) [reply .github/workflows/nodejs.yml]: Fair enough, thanks for the explanation! 

I do have to say though that I hate that it's implicit.

Comment (D4nte) [reply .github/workflows/nodejs.yml]: well, I think it's a npm standard here to allow you to do exactly that.

**Pull request merged**